backend/tests/helpers/query_benchmark/db_query_profiler.py

```diff
@@ -168,9 +181,9 @@ def __exit__(
 
 
 class InfrahubDatabaseProfiler(InfrahubDatabase):
-    def __init__(self, query_analyzer: QueryAnalyzer, **kwargs: Any) -> None:
+    def __init__(self, **kwargs: Any) -> None:
```

Comment:
I think there is some value to have the query_analyzer decoupled from the InfrahubDatabaseProfiler.
In the future we might want to use the same query_analyzer across multiple tests which might result in multiple InfrahubDatabaseProfiler

Reply:
Yes, actually what is confusing with current design is that `QueryAnalyzer` serves to many purposes:
- it contains the information whether `InfrahubDatabaseProfiler` should profile or not
- it stores measurements
- it generates graphes

I think it would be better that two above tasks are performed directly within `InfrahubDatabaseProfiler`. Thus, `InfrahubDatabaseProfiler` is truly responsible of profiling without needing a `QueryAnalyzer`, and `QueryAnalyzer` could become a `GraphGenerator` responsible only for generating graphes and might be shared accross multiple tests.

Reply:
As discussed, will be done in another PR.